drivers/dai/intel/ssp/ssp.c

```diff
@@ -1533,46 +1533,147 @@ static int dai_ssp_set_config_tplg(struct dai_intel_ssp *dp, const struct dai_co
 	return ret;
 }
 
-static int dai_ssp_set_config_blob(struct dai_intel_ssp *dp, const struct dai_config *cfg,
-				    const void *spec_config)
+static int dai_ssp_parse_aux_data(struct dai_intel_ssp *dp, const void *spec_config)
+{
+	const struct dai_intel_ipc4_ssp_configuration_blob_ver_1_5 *blob = spec_config;
+	int aux_tlv_size = sizeof(struct ssp_intel_aux_tlv);
+	int hop, i, j, cfg_len, pre_aux_len, aux_len;
+	struct ssp_intel_aux_tlv *aux_tlv;
+	struct ssp_intel_mn_ctl *mn;
+	struct ssp_intel_clk_ctl *clk;
+	struct ssp_intel_tr_ctl *tr;
+	struct ssp_intel_run_ctl *run;
+	struct ssp_intel_node_ctl *node;
+	struct ssp_intel_sync_ctl *sync;
+	struct ssp_intel_ext_ctl *ext;
+	struct ssp_intel_link_ctl *link;
+	uint8_t *aux_ptr;
+
+	cfg_len = blob->size;
+	pre_aux_len = sizeof(*blob) + blob->i2s_clock_control.mdivrcnt * sizeof(uint32_t);
+	aux_len = cfg_len - pre_aux_len;
+	aux_ptr = (uint8_t *)blob + pre_aux_len;
+
+	if (aux_len <= 0)
+		return 0;
+
+	for (i = 0; i < aux_len; i += hop) {
+		aux_tlv = (struct ssp_intel_aux_tlv *)(aux_ptr);
+		switch (aux_tlv->type) {
+		case SSP_MN_DIVIDER_CONTROLS:
+			mn = (struct ssp_intel_mn_ctl *)&aux_tlv->val;
+			LOG_INF("%s mn div_m %u, div_n %u", __func__, mn->div_m, mn->div_n);
+			break;
+		case SSP_DMA_CLK_CONTROLS:
+			clk = (struct ssp_intel_clk_ctl *)&aux_tlv->val;
+			LOG_INF("%s clk start %u, stop %u", __func__, clk->start, clk->stop);
+			break;
+		case SSP_DMA_TRANSMISSION_START:
+		case SSP_DMA_TRANSMISSION_STOP:
+			tr = (struct ssp_intel_tr_ctl *)&aux_tlv->val;
+			LOG_INF("%s tr sampling frequency %u, bit_depth %u, channel_map %u,",
+				__func__, tr->sampling_frequency, tr->bit_depth, tr->channel_map);
+			LOG_INF("channel_config %u, interleaving_style %u, format %u",
+				tr->channel_config, tr->interleaving_style, tr->format);
+			break;
+		case SSP_DMA_ALWAYS_RUNNING_MODE:
+			run = (struct ssp_intel_run_ctl *)&aux_tlv->val;
+			LOG_INF("%s run enabled %u", __func__, run->enabled);
+			break;
+		case SSP_DMA_SYNC_DATA:
+			sync = (struct ssp_intel_sync_ctl *)&aux_tlv->val;
+			LOG_INF("%s sync sync_denominator %u, count %u", __func__,
+				sync->sync_denominator, sync->count);
+			node = (struct ssp_intel_node_ctl *)((uint8_t *)sync +
+							     sizeof(struct ssp_intel_sync_ctl));
+			for (j = 0; j < sync->count; j++) {
+				LOG_INF("%s node node_id %u, sampling_rate %u", __func__,
+					node->node_id, node->sampling_rate);
+				node++;
+			}
+			break;
+		case SSP_DMA_CLK_CONTROLS_EXT:
+			ext = (struct ssp_intel_ext_ctl *)&aux_tlv->val;
+			LOG_INF("%s ext ext_data %u", __func__, ext->ext_data);
+			break;
+		case SSP_LINK_CLK_SOURCE:
+			link = (struct ssp_intel_link_ctl *)&aux_tlv->val;
+#ifdef CONFIG_SOC_SERIES_INTEL_ACE
```

Comment:
the whole _case SSP_LINK_CLK_SOURCE_ should be in ifdef

Reply:
I would like to get logs anyway if someone is sending this... I mean the enum is valid without ace also

Reply:
Its not really valid, simply does not exist in non-ace closed source and will return error if such config provided.
Not a big deal in this specific field but in general this is error prone if we will accept incorrect config

Reply:
aah you mean this particular enum is evolution in the post-tgl products. ok fine with me, will change the thing.

Reply:
yes, exactly